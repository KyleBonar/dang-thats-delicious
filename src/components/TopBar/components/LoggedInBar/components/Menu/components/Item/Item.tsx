import * as React from "react";
import { Link } from "../../../../../../../Link/Link";
import styled from "../../../../../../../../theme/styled-components";

const StyledDiv = styled.div`
  display: inline-block;
  width: 100%;

  a {
    padding: 8px 12px;
    width: 100%;
    display: block;
    color: ${x => x.theme.primaryThemeColor};
    &:hover,
    &:focus {
      background-color: ${x => x.theme.primaryLightThemeColor};
      color: ${x => x.theme.primaryThemeColor};
    }
  }
`;

export interface ItemProps {
  to?: string;
  children: string;
  onClick?: (event: any) => void;
}

const Item: React.SFC<ItemProps> = props => {
  const _defaultPath = "#";

  return (
    <StyledDiv onClick={props.onClick}>
      <Link href={props.to || _defaultPath}>{props.children}</Link>
    </StyledDiv>
  );
};

export default Item;
