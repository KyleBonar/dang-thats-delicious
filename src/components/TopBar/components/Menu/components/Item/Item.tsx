import * as React from "react";
import { Link } from "../../../../../Link/Link";
import styled from "styled-components";

const StyledDiv = styled.div`
  display: inline-block;
  width: 100%;

  a {
    padding: 8px 12px;
    width: 100%;
    display: block;
    color: ${x => x.theme.primaryThemeColor};
    &:hover,
    &:focus {
      background-color: ${x => x.theme.primaryLightThemeColor};
      color: ${x => x.theme.primaryThemeColor};
    }
  }
`;

interface ItemProps {
  to?: string;
  children: string;
  onClick?: (event: any) => void;
}

const Item: React.SFC<ItemProps> = props => {
  return (
    <StyledDiv onClick={props.onClick}>
      <Link to={props.to || "#"}>{props.children}</Link>
    </StyledDiv>
  );
};

export default Item;
