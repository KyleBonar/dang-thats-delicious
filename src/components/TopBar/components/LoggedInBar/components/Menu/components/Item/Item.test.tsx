import * as React from "react";
import * as enzyme from "enzyme";

import Item, { ItemProps } from "./Item";
import {
  casual,
  ITERATION_SIZE
} from "../../../../../../../../utils/testUtils/testUtils";

const fakeItem = (): ItemProps => ({
  to: casual.random_element([undefined, casual.url]),
  children: casual.string,
  onClick: jest.fn()
});

const mockItems = new Array(ITERATION_SIZE).fill(null).map(fakeItem);

describe("<Item />", () => {
  const _defaultPath = "#";

  it("renders", () => {
    mockItems.forEach(mockItem => {
      const wrapper = enzyme.shallow(
        <Item {...mockItem}>{mockItem.children}</Item>
      );

      expect(wrapper).toBeTruthy();
    });
  });

  it("matches snapshot", () => {
    mockItems.forEach(mockItem => {
      const wrapper = enzyme.shallow(
        <Item {...mockItem}>{mockItem.children}</Item>
      );

      expect(wrapper).toMatchSnapshot();
    });
  });

  it("renders a Link component", () => {
    mockItems.forEach(mockItem => {
      if (!mockItem.to) return;

      const wrapper = enzyme.shallow(
        <Item {...mockItem}>{mockItem.children}</Item>
      );

      expect(wrapper.find("Link").exists()).toBeTruthy();
    });
  });

  it("passes expected 'to' to Link component when provided", () => {
    mockItems.forEach(mockItem => {
      if (!mockItem.to) return;

      const wrapper = enzyme.shallow(
        <Item {...mockItem}>{mockItem.children}</Item>
      );

      expect(wrapper.find("Link").prop("href")).toEqual(mockItem.to);
    });
  });

  it("passes default 'to' to Link component when not provided", () => {
    mockItems.forEach(mockItem => {
      if (mockItem.to) return;

      const wrapper = enzyme.shallow(
        <Item {...mockItem}>{mockItem.children}</Item>
      );

      expect(wrapper.find("Link").prop("href")).toEqual(_defaultPath);
    });
  });

  it("passes expected children to Link component", () => {
    mockItems.forEach(mockItem => {
      const wrapper = enzyme.shallow(
        <Item {...mockItem}>{mockItem.children}</Item>
      );

      expect(wrapper.find("Link").prop("children")).toEqual(mockItem.children);
    });
  });
});
