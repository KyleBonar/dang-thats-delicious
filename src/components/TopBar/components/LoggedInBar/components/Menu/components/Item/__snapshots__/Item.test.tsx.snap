// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> matches snapshot 1`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="#"
  >
    quidem praesentium magnam fugiat error molestiae laborum
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 2`] = `
<styled.div>
  <Link
    href="#"
  >
    facilis ut explicabo iure vitae occaecati officia
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 3`] = `
<styled.div>
  <Link
    href="http://www.Yundt.tv/"
  >
    reprehenderit soluta non omnis consequatur enim quia
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 4`] = `
<styled.div>
  <Link
    href="#"
  >
    reprehenderit omnis adipisci eius laborum eius aut
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 5`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="#"
  >
    delectus vel ipsa tempora aut sint aperiam
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 6`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="#"
  >
    perferendis dolorum cupiditate illum saepe ut assumenda
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 7`] = `
<styled.div>
  <Link
    href="http://www.Greenholt.org/"
  >
    exercitationem est sapiente laudantium molestias assumenda quasi
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 8`] = `
<styled.div>
  <Link
    href="http://Weissnat.name/"
  >
    rem similique temporibus aliquam sed sapiente soluta
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 9`] = `
<styled.div>
  <Link
    href="#"
  >
    nemo doloremque maiores esse atque et non
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 10`] = `
<styled.div>
  <Link
    href="http://www.Eva.info/"
  >
    voluptas ea ducimus autem sit aspernatur aspernatur
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 11`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="#"
  >
    quam nulla est unde sit ex expedita
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 12`] = `
<styled.div>
  <Link
    href="#"
  >
    suscipit omnis necessitatibus maiores ullam quae corporis
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 13`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="http://Smitham.biz/"
  >
    qui enim et doloremque voluptate in sunt
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 14`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="http://Ashton.org/"
  >
    at sint dolores fugiat quis et voluptas
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 15`] = `
<styled.div>
  <Link
    href="http://Zulauf.tv/"
  >
    quis quae iusto sapiente nostrum aut corrupti
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 16`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="#"
  >
    sint officiis qui dignissimos dolore qui voluptatem
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 17`] = `
<styled.div>
  <Link
    href="#"
  >
    similique et maxime quia dolores quisquam aut
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 18`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="http://Wolf.ca/"
  >
    autem quam qui facere in ut odit
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 19`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="#"
  >
    tempora at modi occaecati illum laborum quos
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 20`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="http://www.Sofia.org/"
  >
    consequatur itaque et in fugit in rerum
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 21`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="http://www.Dayana.name/"
  >
    inventore aut eveniet beatae incidunt ut ducimus
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 22`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="http://Rachelle.io/"
  >
    libero aut autem ut ullam et possimus
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 23`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="#"
  >
    quos minus deserunt nihil quasi minus quo
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 24`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="http://McLaughlin.net/"
  >
    et optio consequatur velit voluptate necessitatibus excepturi
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 25`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="#"
  >
    atque quos rerum nobis nam dolor quas
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 26`] = `
<styled.div>
  <Link
    href="#"
  >
    occaecati et consequatur consectetur sunt molestiae cupiditate
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 27`] = `
<styled.div>
  <Link
    href="http://www.Bailey.biz/"
  >
    sint et ipsa consequuntur quidem dolorem quia
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 28`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="http://Okey.name/"
  >
    quis quis nobis repudiandae aliquam molestiae occaecati
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 29`] = `
<styled.div>
  <Link
    href="http://Price.me/"
  >
    blanditiis in eum ut dignissimos aspernatur voluptate
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 30`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="#"
  >
    quia aut sint cumque sunt mollitia quibusdam
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 31`] = `
<styled.div
  onClick={[MockFunction]}
>
  <Link
    href="#"
  >
    quia assumenda ut neque saepe est labore
  </Link>
</styled.div>
`;

exports[`<Item /> matches snapshot 32`] = `
<styled.div>
  <Link
    href="http://Ronny.info/"
  >
    nihil repudiandae aliquid ex architecto ut corrupti
  </Link>
</styled.div>
`;
